src/ORM/Association/ExternalAssociationTrait.php

```diff
@@ -57,7 +57,7 @@ public function foreignKey($key = null)
     {
         if ($key === null) {
             if ($this->_foreignKey === null) {
-                $this->_foreignKey = $this->_modelKey($this->source()->alias());
+                $this->_foreignKey = $this->_modelKey($this->source()->table());
```

Comment:
My comment was more about BC than anything else. Changing this I believe could have negative effects that are just not covered by current test suite.